strawberryfields/utils.py

```diff
@@ -140,9 +141,11 @@ def coherent_state(a, basis="fock", fock_dim=5, hbar=2.0):
     Returns:
         array: the coherent state
     """
+    a = r * np.exp(1j * phi)
+
     if basis == "fock":
         state = np.array(
-            [np.exp(-0.5 * np.abs(a) ** 2) * a ** n / np.sqrt(fac(n)) for n in range(fock_dim)]
+            [np.exp(-0.5 * r ** 2) * a ** n / np.sqrt(fac(n)) for n in range(fock_dim)]
```

Comment:
This can explode easily for `n`~100 can be easily solved using a recursion. However it makes more sense to fix it in a different PR.